### PR TITLE
Convert LanguageFragmentTest.java into kotlin

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes: #3046 
Parent Issue: #1826 

Converted LanguageFragmentTest.java into kotlin

### CODE REVIEW

**Review by gouri-panda — COMMENTED** [review]
@MohitMaliDeveloper This class has already broken. The tests are ignored. We should fix this or create a new ticket to fix this.
cc @kelson42

**Comment by kelson42** [comment]
@MohitMaliFtechiz can we fix it please? or is rhat a big task? what is the problem here?

**Comment by gouri-panda** [comment]
>@MohitMaliFtechiz can we fix it please? or is rhat a big task? what is the problem here?

@kelson42 I think it needs another specific ticket. It's a big task. AFAIK it broke after we migrate to one Activity. Lots of code and UI refactored. I think it would be best if we write it from the scratch and get rid of this class. It doesn't do anything.

File: app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.kiwix.kiwixmobile.testutils.TestUtils
+
+@RunWith(AndroidJUnit4::class)
+class LanguageFragmentTest {
+  @Rule
+  var activityTestRule = ActivityTestRule(
+    KiwixMainActivity::class.java
```
[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.kt]: hi @gouri-panda 

done

File: app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  @Rule
+  var writePermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+
+  @Before fun setUp() {
+    Intents.init()
+  }
+
+  @Test
+  @Ignore("Covert to kotlin/robot")
```
[the remark this answers is not in the document]
Comment (MohitMaliFtechiz) [reply app/src/androidTest/java/org/kiwix/kiwixmobile/language/LanguageFragmentTest.kt]: hi @gouri-panda i have re-enable the test, please re-review this PR.

**Comment by gouri-panda** [comment]
@kelson42 @MohitMaliDeveloper What should we do with this PR? IMO fixing this PR will cost more or should I proceed with this for now?

**Comment by kelson42** [comment]
@MohitMaliFtechiz Your opinion?

**Comment by MohitMaliFtechiz** [comment]
> @kelson42 @MohitMaliDeveloper What should we do with this PR? IMO fixing this PR will cost more or should I proceed with this for now?

hi @gouri-panda , I have enabled test for this , let me know if any other functionality to be tested on this ?

**Review by gouri-panda — APPROVED** [review]
@MohitMaliDeveloper OK, We should create another ticket for the remaining test cases.

**Comment by MohitMaliFtechiz** [comment]
> 

Yes sure

> @MohitMaliDeveloper OK, We should create another ticket for the remaining test cases.

yes sure, please create a ticket for remaining test cases.

**Comment by kelson42** [comment]
ok, I will merge this when this is done, to be sure we don't forget.

**Comment by kelson42** [comment]
@gouri-panda Can you please open the ticket and link here?

**Comment by gouri-panda** [comment]
@kelson42 #3121

**Pull request merged**